app/src/androidTest/java/com/example/sharingang/FakeCurrentUserProvider.kt

```diff
@@ -19,15 +20,27 @@ class FakeCurrentUserProvider @Inject constructor(
     }
 
     override fun getCurrentUserId(): String? {
-        return if (instance == 1) fakeUser1.id else fakeUser2.id
+        return when (instance) {
+            1 -> fakeUser1.id
+            2 -> fakeUser2.id
+            else -> null
+        }
     }
 
     override fun getCurrentUserEmail(): String {
-        return "test-user@example.com"
+        return if(instance != 0) "test-user@example.com" else ""
+    }
+
+    override fun getCurrentUserName(): String {
+        return when (instance) {
+            1 -> fakeUser1.name
+            2 -> fakeUser2.name
+            else -> ""
+        }
     }
 
-    override fun getCurrentUserName(): String? {
-        return if(instance == 1) fakeUser1.name else fakeUser2.name
+    override fun getCurrentUser(): FirebaseUser? {
+        return null
```

Comment:
It's because we are implementing CurrentUserProvider, since we need to override methods, we need to implement this one in FakeCurrentUserProvider. As it is a fake user provider, we do not provide a Firebase user, so we just return null.

Reply:
Perfect, thanks for the explanation:)